osu.Game.Rulesets.Taiko/Difficulty/TaikoPerformanceCalculator.cs

```diff
@@ -14,37 +14,35 @@ namespace osu.Game.Rulesets.Taiko.Difficulty
 {
     public class TaikoPerformanceCalculator : PerformanceCalculator
     {
-        protected new TaikoDifficultyAttributes Attributes => (TaikoDifficultyAttributes)base.Attributes;
-
-        private Mod[] mods;
         private int countGreat;
         private int countOk;
         private int countMeh;
         private int countMiss;
 
-        public TaikoPerformanceCalculator(Ruleset ruleset, DifficultyAttributes attributes, ScoreInfo score)
-            : base(ruleset, attributes, score)
+        public TaikoPerformanceCalculator(Ruleset ruleset)
+            : base(ruleset)
```

Comment:
I'm wondering if this needs the ruleset input. ie. is there any reason to not do:

```suggestion
        public TaikoPerformanceCalculator() : base(new TaikoRuleset())
```

Reply:
Sure

Reply:
Also removed the similar argument in the ScoreProcessor PR in https://github.com/ppy/osu/pull/17243/commits/daac93349864ee975aab450fd6b7d4b48e77bac6

Reply:
I'm not 100% on this change but I guess it's probably fine as long as `Ruleset` doesn't contain any dynamic pieces going forward (and as long as we don't mind one extra `RulesetInfo` being constructed). Let's see how things pan out.